drift/lib/src/runtime/manager/manager.dart

```diff
@@ -450,50 +456,49 @@ abstract class BaseTableManager<
   ///
   /// Uses the distinct flag to ensure that only distinct rows are returned
   @override
-  Stream<DT?> watchSingleOrNull() => $state
-      .copyWith(distinct: true)
+  Stream<$Dataclass?> watchSingleOrNull({bool distinct = true}) => $state
+      .copyWith(distinct: distinct)
       .buildSelectStatement()
       .watchSingleOrNull();
 }
 
 /// A table manager that exposes methods to a table manager that already has filters/orderings/limit applied
 //  As of now this is identical to [BaseTableManager] but it's kept seperate for future extensibility
+@immutable
 class ProcessedTableManager<
```

Comment:
We're currently generating typedefs to this class, but these aren't being referenced by generated code elsewhere.

So I'm wondering if we even need to have this class at the moment? If we have it to add new features in the future like the documentation suggest, shouldn't it be referenced in code? I understand if we explicitly reference `RootTableManager` now and then realize that we should have been referencing a more general manager class instead, we have a problem - and that justifies having `ProcessedTableManager` around.
But as far as I can see, we're not currently using this class outside of deaf typedef. So what's the point of having it? Given that it's not currently used, couldn't we just introduce separate classes when we need them and only start referencing them then? Or am I overlooking something here?

Reply:
Ah I see we're using it as a return value - that's fair enough, but can we remove the typedef in generated code?

Reply:
That makes total sense, thanks for the detailed explanation!

I have added a brief section about the `filter` + `create` incompatibility as a motivating example to the `ProcessedTableManager` docs.